tests/templates/test_layers/test_simplified_twodesign.py

```diff
@@ -0,0 +1,336 @@
+        ],
+    )
+    def test_shape(self, n_layers, n_wires, expected_shape):
+        """Test that the shape method returns the correct shape of the weights tensor"""
+
+        shape = qml.templates.SimplifiedTwoDesign.shape(n_layers, n_wires)
+        assert shape == expected_shape
+
+
+def circuit_template(initial_weights, weights):
+    qml.templates.SimplifiedTwoDesign(initial_weights, weights, range(3))
+    return qml.expval(qml.PauliZ(0))
+
+
+def circuit_decomposed(initial_weights, weights):
+    qml.RY(initial_weights[0], wires=0)
+    qml.RY(initial_weights[1], wires=1)
+    qml.RY(initial_weights[2], wires=2)
+
+    qml.CZ(wires=[0, 1])
+    qml.RY(weights[0, 0, 0], wires=0)
+    qml.RY(weights[0, 0, 1], wires=1)
+
+    qml.CZ(wires=[1, 2])
+    qml.RY(weights[0, 1, 0], wires=1)
+    qml.RY(weights[0, 1, 1], wires=2)
+
+    return qml.expval(qml.PauliZ(0))
+
+
+class TestInterfaces:
+    """Tests that the template is compatible with all interfaces, including the computation
+    of gradients."""
+
+    def test_list_and_tuples(self, tol):
+        """Tests common iterables as inputs."""
+
+        weights = [[[[0.1, -1.1], [0.2, 0.1]], [[0.1, -1.1], [0.2, 0.1]]]]
+        initial_weights = [0.1, 0.2, 0.3]
+
+        dev = qml.device("default.qubit", wires=3)
+
+        circuit = qml.QNode(circuit_template, dev)
+        circuit2 = qml.QNode(circuit_decomposed, dev)
+
+        res = circuit(initial_weights, weights)
+        res2 = circuit2(initial_weights, weights)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+        weights_tuple = [[tuple(weights[0][0]), tuple(weights[0][1])]]
+        init_weights_tuple = tuple(initial_weights)
+        res = circuit(init_weights_tuple, weights_tuple)
+        res2 = circuit2(init_weights_tuple, weights_tuple)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+    def test_autograd(self, tol):
+        """Tests the autograd interface."""
+
+        weights = np.random.random(size=(1, 2, 2))
+        weights = pnp.array(weights, requires_grad=True)
+        initial_weights = np.random.random(size=(3,))
+        initial_weights = pnp.array(initial_weights, requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=3)
+
+        circuit = qml.QNode(circuit_template, dev)
+        circuit2 = qml.QNode(circuit_decomposed, dev)
+
+        res = circuit(initial_weights, weights)
+        res2 = circuit2(initial_weights, weights)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+        grad_fn = qml.grad(circuit)
+        grads = grad_fn(initial_weights, weights)
+
+        grad_fn2 = qml.grad(circuit2)
+        grads2 = grad_fn2(initial_weights, weights)
+
+        assert np.allclose(grads[0], grads2[0], atol=tol, rtol=0)
+        assert np.allclose(grads[1], grads2[1], atol=tol, rtol=0)
+
+    def test_jax(self, tol, skip_if_no_jax_support):
+        """Tests the jax interface."""
+
+        import jax
+        import jax.numpy as jnp
+
```

Comment:
Be sure to skip the test if JAX isn't installed.

Reply:
Oh I forgot that too, thanks!